src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountTests.fs

```diff
@@ -0,0 +1,115 @@
+module BankAccountTests
+
+open System
+open Deciders.BankAccount
+open Deciders.BankAccountPrimitives
+open FsCheck
+open Ogooreck.BusinessLogic
+open Deciders.BankAccountDecider
+open FsCheck.Xunit
+
+let random = Random()
+let now = DateTimeOffset.UtcNow
+
+let spec =
+    Specification.For(decide, evolve, (fun () -> NotInitialised))
```

Comment:
Bad idea to allow effects into the creation of the `initial` state - just ask for a value and remove the `fun ()`

Reply:
Extended API in https://github.com/oskardudycz/Ogooreck/pull/13/commits/109632e057c4b29ab35d4d14669f3849c475195b 🙂